### PR TITLE
Add Github CI Workflows

### PR DESCRIPTION
**Pull request by khuang93**

### CODE REVIEW

File: .github/workflows/build_test_1804.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Build and Test on Ubuntu 18.04
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
```
Comment (souryavarenya) [inline .github/workflows/build_test_1804.yml:6]: Is this on merge or on create?

File: .github/workflows/build_test_1804.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
+
+    strategy:
+      matrix:
+        cmake-version: [3.12]
```
Comment (souryavarenya) [inline .github/workflows/build_test_1804.yml:16]: Internally we use 3.18. 3.12 shouldn't be an issue though.

File: .github/workflows/build_test_2004.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      matrix:
+        cmake-version: [3.12]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Deps
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libeigen3-dev libyaml-cpp-dev libgtest-dev
+    # - name: Install Gtest
```
Comment (souryavarenya) [inline .github/workflows/build_test_2004.yml]: Remove?

File: .github/workflows/build_test_1804.yml (hunk 1 above)
Comment (khuang93) [reply .github/workflows/build_test_1804.yml:6]: create AFAIK

**Pull request merged**